app/controllers/articles_controller.rb

```diff
@@ -185,6 +185,19 @@ def stats
     @organization_id = @article.organization_id
   end
 
+  def admin_unpublish
+    authorize @article
+    if @article.has_frontmatter?
+      @article.body_markdown.sub!("\npublished: true\n", "\npublished: false\n")
+    else
+      @article.published = false
+    end
+
+    return unless @article.save
+
+    redirect_to "#{URL.url@article.current_state_path}"
```

Comment:
I'm not sure if this solution is exactly what you're looking for, but this redirect opens the article editor in the Mod Actions Panel giving the Admin the ability to edit the article from there once it is unpublished. I found this redirect in the #update method in this controller (`Articles::Controller`) for reference! 😄 
```suggestion
    redirect_to("#{@article.path}/edit")
```

Reply:
Thanks @juliannatetreault! I'm not sure if the article's `edit` path is where we want to end up though.
The bigger issue is getting the redirect to happen within the actual article page, not within the iframe 😓 
Thank you though!

Reply:
@msarit Iframes can't update the location of the embedding site by default for security reasons. You need to add an option to your `<iframe>` tag:

```
sandbox="allow-top-navigation"
```